src/main/java/org/jenkinsci/plugins/bigpanda/ChangesBuilder.java

```diff
@@ -0,0 +1,98 @@
+package org.jenkinsci.plugins.bigpanda;
+
+import java.util.List;
+
+import hudson.model.Cause;
+import hudson.model.Job;
+import hudson.model.Result;
+import hudson.model.Run;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+
+/**
+ * Will convert the build data into a JSON object to send to BigPanda
+ */
+public class ChangesBuilder {
+
+    private Run<?, ?> build;
+
+    public ChangesBuilder(Run<?, ?> build) {
+        this.build = build;
+    }
+
+    public JSONObject create() {
+        JSONObject json = new JSONObject();
+        Result result = this.build.getResult();
+
+        // map status
+        if (this.build.isBuilding()) {
+            json.put("status", "In Progress");
+        } else if (result == Result.FAILURE || result == Result.ABORTED || result == Result.NOT_BUILT) {
+            json.put("status", "Canceled");
+        } else {
+            json.put("status", "Done");
+        }
+
+        json.put("identifier", String.valueOf(this.build.getDisplayName()));
+        json.put("start", (this.build.getStartTimeInMillis() / 1000));
+        json.put("summary", this.build.getFullDisplayName().replaceAll("[^a-zA-Z0-9-# ]", ""));
+        json.put("ticket_url", Jenkins.getInstance().getRootUrl() + this.build.getUrl());
```

Comment:
IIRC there's a get absolute url for the build

Reply:
According to Jenkins docs, that method is deprecated

https://javadoc.jenkins-ci.org/hudson/model/Run.html#getAbsoluteUrl--